blockchain/utxocache.go

```diff
@@ -0,0 +1,1085 @@
+// Copyright (c) 2015-2018 The btcsuite developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+package blockchain
+
+import (
+	"container/list"
+	"fmt"
+	"sync"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/database"
+	"github.com/btcsuite/btcd/txscript"
+	"github.com/btcsuite/btcd/wire"
+)
+
+const (
+	// utxoBatchSizeBlocks is the maximum number of blocks to be processed in a
+	// single transaction.
+	utxoBatchSizeBlocks = 50
+
+	// utxoFlushPeriodicThreshold is the threshold percentage at which a flush is
+	// performed when the flush mode FlushPeriodic is used.
+	utxoFlushPeriodicThreshold = 90
+
+	// outpointSize is the size of an outpoint in a 64-bit system.
+	//
+	// This value is calculated by running the following on a 64-bit system:
+	//	unsafe.Sizeof(wire.OutPoint{})
```

Comment:
nit: Index is explicitly int32, so 64-bit system or not doesn't matter IIUC.

Reply:
Had to look up some docs to make sure.

I think you're right. [32]byte should be 32 on 32bit systems and uint32 should be 4. I'll remove the comments like these in other parts of the code as well.